test/PR_test/unit_test/dataset/test_extend_dataset.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2020 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import unittest
+
+import numpy as np
+
+import fastestimator as fe
+from fastestimator.dataset import GeneratorDataset
+
+
+def inputs():
+    while True:
+        yield {'x': np.random.rand(16), 'y': np.random.randint(16)}
+
+
+class TestBatchDataset(unittest.TestCase):
+    def test_dataset_contraction(self):
+        ds1 = GeneratorDataset(generator=inputs(), samples_per_epoch=10)
+        unpaired_ds = fe.dataset.ExtendDataset(dataset=ds1, spoof_length=5)
+        pipeline = fe.Pipeline(unpaired_ds)
```

Comment:
what happens when a dataset has, for example, 23 data points, batch size 25, drop_last=True and then the user extends the ds?

Reply:
'y': tensor([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,18, 19, 20, 21, 22,  0,  1])

This is the result of a dataset with 23 samples and batch size of 25, with drop_last=True and spoof_length of 30